asreview/webapp/api/auth.py

```diff
@@ -0,0 +1,512 @@
+# Copyright 2019-2022 The ASReview Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import datetime as dt
+import smtplib
+import ssl
+from pathlib import Path
+
+from flask import Blueprint
+from flask import current_app
+from flask import jsonify
+from flask import render_template_string
+from flask import request
+from flask_cors import CORS
+from flask_login import current_user
+from flask_login import login_user
+from flask_login import logout_user
+from flask_mail import Mail
+from flask_mail import Message
+from sqlalchemy import and_
+from sqlalchemy import or_
+from sqlalchemy.exc import IntegrityError
+from sqlalchemy.exc import SQLAlchemyError
+
+from asreview.webapp import DB
+from asreview.webapp.authentication.login_required import asreview_login_required  # NOQA
+from asreview.webapp.authentication.models import User
+from asreview.webapp.authentication.oauth_handler import OAuthHandler
+
+# TODO: I need a folder to stash templates for emails,
+# is this the way we should do it? I don't see the point
+# of making the end-user decide the exact location.
+bp = Blueprint("auth", __name__, url_prefix="/auth")
+
+CORS(
+    bp,
+    resources={r"*": {"origins": ["http://localhost:3000", "http://127.0.0.1:3000"]}},
+    supports_credentials=True,
+)
+
+
+def perform_login_user(user):
+    """Helper function to login a user"""
+    return login_user(user, remember=True, duration=dt.timedelta(days=31))
+
+
+# TODO: not sure if this file is the right place for this function
+def send_email(msg, config):
+    context = ssl.create_default_context()
+    with smtplib.SMTP_SSL(config["SERVER"], config["PORT"], context=context) as server:
+
+        # login smtp server
+        server.login(config["USERNAME"], config["PASSWORD"])
+        # send email
+        return server.sendmail(msg)
+
+
+# TODO: not sure if this file is the right place for this function
+def send_forgot_password_email(user, cur_app):
+    # get necessary information out of user object
+    name = user.name or "ASReview user"
+    # email config
+    config = cur_app.config.get("EMAIL_CONFIG")
+    # TODO: this is horrible => what if there is a domain name,
+    # where is it coming from? Where can I get it?
+    root_url = "http://127.0.0.1:3000/"
```

Comment:
Can this be taken from the config? Not sure what is the right way. Maybe @J535D165 knows?